ambry-api/src/main/java/com.github.ambry/store/Transformer.java

```diff
@@ -26,6 +28,14 @@
    * @return the output {@link TransformationOutput}.
    */
   TransformationOutput transform(Message message);
+
+  /**
+   * Warms up transformer with message infos representing messages
+   * it will transform later
+   * @param messageInfos
+   * @throws Exception
```

Comment:
this is too broad. `transform()` doesn't throw an exception, I don't think this should either (runtime exceptions ok)

Reply:
So instead it should... return an Exception object?  Return a TransformationOutput that may have an Exception or an... empty Message object that means nothing?  idk seems like throwing an exception makes the most sense